### PR TITLE
Add composition root

### PR DESCRIPTION
**Pull request by janhohenheim**

### CODE REVIEW

File: LinqToRest.Test/QueryProviderBuilderTest.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            var queryProvider = builder
+                .HttpClient(httpClient)
+                .Root(new Uri("https://www.example.com"))
+                .Build();
+            
+            Assert.NotNull(queryProvider);
+        }
+        
+    }
+}
```
Comment (dngroth) [inline LinqToRest.Test/QueryProviderBuilderTest.cs]: Terminating newline

File: LinqToRest/QueryProviderBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            var entityValidator = CreateEntityValidator();
+            return () => new QueryBinder(entityValidator);
+        }
+
+        private static EntityValidator CreateEntityValidator()
+        {
+            return new EntityValidator();
+        }
+    }
+}
```
Comment (dngroth) [inline LinqToRest/QueryProviderBuilder.cs]: Terminating newline

File: LinqToRest/QueryProviderBuilderException.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+            : base(message)
+        {
+        }
+
+        public QueryProviderBuilderException(string message, Exception inner)
+            : base(message, inner)
+        {
+        }
+    }
+}
```
Comment (dngroth) [inline LinqToRest/QueryProviderBuilderException.cs]: Terminating newline

**Review by dngroth — CHANGES_REQUESTED** [review]

File: LinqToRest.Test/QueryProviderBuilderTest.cs (hunk 1 above)
Comment (janhohenheim) [reply LinqToRest.Test/QueryProviderBuilderTest.cs]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/17/commits/4389e5c27fad7d25a9309580a00d534b44f25f2f

File: LinqToRest/QueryProviderBuilder.cs (hunk 2 above)
Comment (janhohenheim) [reply LinqToRest/QueryProviderBuilder.cs]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/17/commits/4389e5c27fad7d25a9309580a00d534b44f25f2f

File: LinqToRest/QueryProviderBuilderException.cs (hunk 3 above)
Comment (janhohenheim) [reply LinqToRest/QueryProviderBuilderException.cs]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/17/commits/4389e5c27fad7d25a9309580a00d534b44f25f2f

**Review by dngroth — APPROVED** [review]

**Pull request merged**